### ISSUE TITLE
Create control block reference for ExtRef

### ISSUE THREAD

**Issue by danyill** [OpenEnergyTools/scl-lib#5]
A useful helper function for me would be creating a control block reference for an ExtRef.

I use code something like this at the moment:

```ts
export function getCbReference(extRef: Element): string {
  const extRefValues = ['iedName', 'srcPrefix', 'srcCBName'];
  const [srcIedName, srcPrefix, srcCBName] = extRefValues.map(
    attr => extRef.getAttribute(attr) ?? ''
  );

  const srcLDInst =
    extRef.getAttribute('srcLDInst') ?? extRef.getAttribute('ldInst');
  const srcLNClass = extRef.getAttribute('srcLNClass') ?? 'LLN0';

  return `${srcIedName}${srcPrefix}${srcLDInst}/${srcLNClass}.${srcCBName}`;
}
```

I would be happy to tidy this up, add tests and do a PR if this seems reasonable, in a similar manner to `tControl/controlBlockObjRef.ts`.

**Comment by JakobVogelsang** [comment]
I agree with you and would propose something similar. There is a function in there `tControl/sourceControlBlock`. When we add tests for this one and export it, you could reach the same goal through

```ts
const objRef = controlBlockObjRef(sourceControlBlock(extRef))
```

That would have the advantage that there is an extra function available that I think might be very useful for others. In general, that is the tricky question we have to answer in this repo. Do we want to have only specialized functions or a set of very general functions that combined properly do the same thing as the generalized function.

**Comment by danyill** [comment]
>  In general, that is the tricky question we have to answer in this repo. Do we want to have only specialized functions or a set of very general functions that combined properly do the same thing as the generalized function.

It is  tricky! I think we want general functions and a "sane approach" to compose them of more specialised functions. As the need arises we refactor the more general ones and any specialised ones they are composed of and expose them for specific use cases. But we should not over-expose the interface without a use case to try and keep the API as general and user-focussed as possible for as long as possible

**Comment by JakobVogelsang** [comment]
In this specific case. Would you rather prefer us having and `getCbReference` function or would it also be ok for you to work `sourceControlBlock` in combination with `controlBlockObjRef`?

**Comment by trusz** [comment]
not to hijack the issue, but I am currently doing it like this:
```txt
┌───────────────┐
│  scd-queries  │
└───────────────┘
        ▲        
        │        
┌───────────────┐
│   use-cases   │
└───────────────┘
        ▲        
        │        
┌───────────────┐
│    client     │
└───────────────┘
```

- `scd-queries` are low level XML queries and helps me navigate the xml
- `use-cases` are higher level and prepare the data for the client
- `client` would be something like an OpenSCD plugin

_an example:_
```txt
              ┌───────────────┐            
              │  scd-queries  │            
              └──┬────────┬───┘            
                 │        │                
     searchGSEControlElements              
                 │        │                
              searchSampledValueControl    
                 │        │                
                 ◠        ◠                
                 ●        ●                
      ┌ ─ ─ ─ ─ ─│─ ─ ─ ─ ┼ ─ ─ ─ ─ ─      
       use-cases │        │          │     
      │    ┌─────┴────────────────┐        
           │       messages       │  │     
      │    └────┬────────────┬────┘        
       ─ ─ ─ ─ ─│─ ─ ─ ─ ─ ─ ┼ ─ ─ ─ ┘     
                │            │             
 findPublishedGOOSEMessagesByIED           
                │            │             
                │ findSampledValuesByIED   
                ◠            │             
                ●            │             
                │            ◠             
                │            ●             
                │            │             
           ┌────┴────────────┴─────────┐   
           │          client           │   
           └───────────────────────────┘   
```

1. here the client needs to find all the GOOSE messages published by an IED. The client is not necessarily need the XML element or even need to know about it. It just needs a well prepared object with all information in it. 
2. the `messages` use case can use any queries it needs to find all the information needs. So it is a bit like an aggregation layer.
3. the `searchGSEControlElement` knows how to look for the XML elements.

So `scd-queries` are like data access layer in classical three layered architecture, `messages` would be the frontend facing API layer and the `client` is the client

What do you think?
